### PR TITLE
Disposebag minor clean up

### PR DESCRIPTION
**Pull request by mayojava**

### CODE REVIEW

**Comment by mayojava** [comment]
I just moved some things around and minor clean ups

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposableExtensions.kt
Comment (kizitonwose) [inline disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposableExtensions.kt]: I don't think this extension belongs in the same scope as the Lifecycle owner extension. They're all Disposable extensions but keeping it in the same file with the DisposeBag class is better, for visibility.

Comment (kizitonwose) [inline disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposableExtensions.kt:8]: I don't think this extension belongs in the same scope as the Lifecycle owner extension. They're all Disposable extensions but keeping it in the same file with the DisposeBag class is better, for visibility.

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt
Comment (kizitonwose) [inline disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt:18]: Any specific reason why the fields have to come before the secondary constructor?

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt
Comment (kizitonwose) [inline disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt:43]: This can be merged but will cause issues for people who already specify the parameter name when calling the method, making it breaking change. Since the method takes only one parameter, I think we can retain the old `d` parameter name which is from the `DisposableContainer` interface method, maybe until the next major release.

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt
Comment (kizitonwose) [inline disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt:55]: The code follows the [official Kotlin Style guide for Android](https://android.github.io/kotlin-guides/style.html#braces), hence braces are not really needed in this case.

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBagPlugins.kt
Comment (kizitonwose) [inline disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBagPlugins.kt:5]: I don't really see the need for a new file just for the Plugins since there isn't much in it right now. I think keeping everything that has to do with `DisposeBag` class in the `DisposeBag` file is better, for visibility.

File: disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt
Comment (kizitonwose) [inline disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt:22]: This `disposeBag` is used in only one test case, I really don't see the need for a field reference.

File: disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt
Comment (kizitonwose) [inline disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt:90]: This test is not meant to check that items are emitted, that is already covered. This is only meant to check that the disposable is properly disposed.

**Review by kizitonwose — CHANGES_REQUESTED** [review]
This PR has many nice changes but some things have to be fixed before it can be merged. I've added comments to the parts that need attention.

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposableExtensions.kt
Comment (mayojava) [reply disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposableExtensions.kt:8]: I think the major thing here should not be visibility but just pure separation of concerns. I see no reason why extension functions should be sitting in same file as main `DisposeBag` class. Doing it this way allows pure separation of concern. And it sorts in a way ensures that that file has just a single reason to change.

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt
Comment (mayojava) [reply disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt:18]: This is just sheepishly borrowed from java convention and standards where you have your fields before the constructors and not the other way round. This is subjective it is okay if you dont want it this way

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt
Comment (mayojava) [reply disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt:43]: I get your point here, if you think there are people already using the library then this will be a valid concern. But then the update will be in different update and will only break for those that upgrade to new version of the library.

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt
Comment (mayojava) [reply disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBag.kt:55]: Yeah this is subjective

File: disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBagPlugins.kt
Comment (mayojava) [reply disposebag/src/main/kotlin/com/kizitonwose/android/disposebag/DisposeBagPlugins.kt:5]: Like i commented earlier it should not be about visibility but separation of concerns. Despite the fact that this file is small, we should not wait till we need to extend it before we separate files. Doing this from the onset shows intent, we easily add to this file in the future without refactoring.

File: disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt
Comment (mayojava) [reply disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt:22]: The reason why I did this is just to explicitly have the class under test as a field. I guess it makes it clearer the object we are testing.

File: disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt
Comment (mayojava) [reply disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt:90]: I understand your point but you still to be exhaustive, event hough that is not what we explicitly testing for, I think just for the purpose of correctness this still needs to checked.

**Comment by mayojava** [comment]
Really most of my comments are subjective and you have the final say here, so its cool if you dont agree with some or all of my changes.

**Comment by kizitonwose** [comment]
Some of the changes in this PR are okay, however, due to the structure of the commits, this PR will not be merged. Some things will be borrowed from this PR in the future. Thanks for submitting.

**Comment by kizitonwose** [comment]
Some of the changes in this PR are okay. However, due to the structure of the commits, this PR will not be merged. Some things will be borrowed from this PR in the future. Thanks for submitting.

**Pull request closed without merging**
